src/components/PdfButton/PdfButton.tsx

```diff
@@ -35,24 +38,31 @@ const formatFileName = (fileName: string): string => {
     return `${newFileName}_${filenameTimestampNow()}.pdf`;
 };
 
-const PdfButton = <T,>({
-    data,
+const PdfButton = <Data, ErrorType>({
+    fetchDataAndFileName,
     text,
     pdfComponent: PdfComponent,
-    fileName,
-    clickHandler = () => {},
+    onPdfCreationCompleted = () => {},
     formatName = true,
-}: Props<T>): React.ReactElement => {
+    disabled = false,
+    onPdfCreationFailed,
+}: Props<Data, ErrorType>): React.ReactElement => {
+    const onClick = async (): Promise<void> => {
+        const { data, error } = await fetchDataAndFileName();
+        if (error) {
+            onPdfCreationFailed(error);
+            return;
+        }
+        const blob = await pdf(<PdfComponent data={data.data} />).toBlob();
+        error;
```

Comment:
blob line: I guess this is the pdf file in some mystical format such that we can refer to it as a variable in the code. the type of blob is literally Blob. 
error line: No idea, have removed, this can't have been doing anything.